website/static/js/registrationEditorExtensions.js

```diff
@@ -0,0 +1,119 @@
+var $ = require('jquery');
+var URI = require('URIjs');
+var moment = require('moment');
+var ko = require('knockout');
+
+var FilesWidget = require('js/FilesWidget');
+var Fangorn = require('js/fangorn');
+var $osf = require('js/osfHelpers');
+
+var node = window.contextVars.node;
+
+ko.bindingHandlers.osfUploader = {
+    init: function(element, valueAccessor, allBindings, viewModel, bindingContext) {
+        var userClick = false;
+        var fw = new FilesWidget(
+            element.id,
+            node.urls.api + 'files/grid/',
+            {
+                onselectrow: function(item) {
+                    userClick = true;
+                    this.multiselected([item]);
+                    self.preview_value = item.data;
+                    this.path = item.data.path;
+                    self.files = item.data;
+
+                    var tb = this;
+                    var fileurl = "";
+                    if (item.data.kind === "file") {
+                        var redir = new URI(item.data.nodeUrl);
+                        redir.segment("files").segment(item.data.provider).segmentCoded(item.data.path.substring(1));
+                        fileurl = redir.toString() + '/';
+                        $("#scriptName").html(item.data.name);
```

Comment:
@leb2dg This will break if we ever have more than one uploader on a page. You should use knockout bindings in the uploader template to accomplish this.